custom/crediblex/portfolio/account/src/main/java/com/crediblex/fineract/portfolio/account/jobs/executestandinginstructions/CustomExecuteStandingInstructionsTasklet.java

```diff
@@ -117,24 +136,43 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
             }
 
             if (isDueForTransfer && transactionAmount != null && transactionAmount.compareTo(BigDecimal.ZERO) > 0) {
+                dueCount++;
                 final SavingsAccount fromSavingsAccount = null;
                 final boolean isRegularTransaction = true;
                 final boolean isExceptionForBalanceCheck = false;
                 AccountTransferDTO accountTransferDTO = new AccountTransferDTO(transactionDate, transactionAmount,
                         data.getFromAccountType(), data.getToAccountType(), data.getFromAccount().getId(), data.getToAccount().getId(),
-                        data.getName() + " Standing instruction trasfer ", null, null, null, null, data.toTransferType(), null, null,
+                        data.getName() + " Standing instruction transfer ", null, null, null, null, data.toTransferType(), null, null,
                         data.getTransferType().getValue(), null, null, ExternalId.empty(), null, null, fromSavingsAccount,
                         isRegularTransaction, isExceptionForBalanceCheck);
-                final boolean transferCompleted = transferAmountWithPartialPaymentSupport(errors, accountTransferDTO, data.getId(),
-                        transactionAmount);
 
-                if (transferCompleted) {
-                    final String updateQuery = "UPDATE m_account_transfer_standing_instructions SET last_run_date = ? where id = ?";
-                    jdbcTemplate.update(updateQuery, transactionDate, data.getId());
+                try {
+                    BigDecimal finalTransactionAmount = transactionAmount;
+                    transactionTemplate.execute(status -> {
+                        final boolean transferCompleted = transferAmountWithPartialPaymentSupport(errors, accountTransferDTO, data.getId(),
```

Comment:
This makes sense .. what is puzzling me is how the method `transferAmountWithPartialPaymentSupport` will fail. I see it catches all the exceptions

Reply:
If there is any exception, it propagates to the primary transaction manager of springbatch (initiating transaction) and all transactions are rolled back.